dirty_cat/similarity_encoder.py

```diff
@@ -166,7 +169,7 @@ class SimilarityEncoder(_BaseEncoder):
 
     def __init__(self, similarity='ngram', ngram_range=(3, 3),
                  categories='auto', dtype=np.float64,
-                 handle_unknown='ignore', hashing_dim=None, n_prototypes=None):
+                 handle_unknown='ignore', hashing_dim=None, n_prototypes=None, sample_weight=False):
```

Comment:
In terms of usability, I am not very enthusiastic at the idea that we are adding an additional parameter to the similarity encoder. We need to find a good default for this parameter, and maybe just hard-code the value.

Reply:
Let's just hard-code the best option for sample_weight here, unless you see a big issue with it.